### DISCUSSION TITLE
Issue with adding negative sampling with "RandomLinkSplit" for a heterogeneous graph

### DISCUSSION THREAD

**Discussion by HamidHadipour** [pyg-team/pytorch_geometric#5446] [Q&A]
Hi community,
I am new to Pyg, I followed the example code in [https://github.com/pyg-team/pytorch_geometric/blob/master/examples/hetero/hetero_link_pred.py](url) and I need to do link prediction on a heterogeneous graph which is technically a bi-partite graph.
My task is link prediction and I have the dataset of connections or (label = 1) so, I need to do negative sampling. But as time as I add a ratio except "0.0" for the "negative sampling ration" property I get "Assertion error"

torch-geometric==2.1.0

Here is the data definition:
`
data = HeteroData()

data['chem'].num_nodes = len(cp_mapping)  
data['protein'].num_nodes = len(cp_mapping_p)

data['chem', 'CPI', 'protein'].edge_index = edge_index
data['chem', 'CPI', 'protein'].edge_label = edge_label

data['chem'].x = torch.eye(data['chem'].num_nodes)
del data['chem'].num_nodes
data['protein'].x = torch.eye(data['protein'].num_nodes)
del data['protein'].num_nodes
`

Here is the result:
`
HeteroData(
  chem={ x=[1206, 1206] },
  protein={ x=[152, 152] },
  (chem, CPI, protein)={
    edge_index=[2, 20000],
    edge_label=[20000]
  }
)
`

Here is how I use RandomLinkSplit:
`

data = T.ToUndirected()(data)
del data['protein', 'rev_CPI', 'chem'].edge_label 
train_data, val_data, test_data = T.RandomLinkSplit(
        num_val=0.2,
        num_test=0.2,
        neg_sampling_ratio=1.0,
        edge_types=[('chem', 'CPI', 'protein')],
        rev_edge_types=[('protein', 'rev_CPI', 'chem')]
    )(data)

`

![Screenshot (1435)](https://user-images.githubusercontent.com/72678480/190293025-f8fe2f34-0359-44d4-8610-9797790ddf3c.png)

Would you please take a look and correct me, or suggest me a better way?
@rusty1s

**Comment by rusty1s** [comment]
We can only add negative edges in case you want to perform classification, in which case `edge_label` needs to be of type `torch.long`. I assume it is a floating-point type for you.

**Reply by HamidHadipour** [reply]
Thanks for your reply.
Would you take a look again?
Actually, my problem is classification and the labels are just "1".
Here is what I have as edge_label and edge_index:


![Screenshot (1437)](https://user-images.githubusercontent.com/72678480/190551044-6eb814db-9c8d-4420-b532-d1d764cf745a.png)

Thank you.

**Reply by rusty1s** [reply]
Ah, it looks like `edge_label_index` and `edge_label` don't match in size. You need a label for each edge, e.g., `edge_label_index.size(1) == edge_label.size(0)`.
